### PR TITLE
Pan Zoom Revisions: Scale drawn marks when zooming

### PR DESCRIPTION
**Pull request by simensta**
This PR:
- scales drawn marks correctly when zooming in on a subject
- removes some unnecessary state variables, methods, and setTimeout calls.

I think the next step is to move pan & zoom into its own component, but this PR will fix the above issues in the meantime.

### CODE REVIEW

File: app/components/frame-viewer.cjsx [hunk 1]
```diff
@@ -119,10 +116,10 @@ module.exports = React.createClass
               </div>
             </div>
             <div>
-              <button title={"zoom out"} className={"zoom-out fa fa-minus" + if @canZoomOut() then " disabled" else "" } onMouseDown={ @continuousZoom.bind(this, 1.1 ) } />
+              <button title={"zoom out"} className={"zoom-out fa fa-minus" + if @canZoomOut() then " disabled" else "" } onMouseDown={ @continuousZoom.bind(this, 1.1 ) } onMouseUp={@stopZoom} />
```
Comment (eatyourgreens) [inline app/components/frame-viewer.cjsx]: Confused by this. Disable the button if you can zoom out?

File: app/components/frame-viewer.cjsx [hunk 2]
```diff
@@ -119,10 +116,10 @@ module.exports = React.createClass
...
             </div>
             <div>
-              <button title={"zoom in"} className="zoom-in fa fa-plus" onMouseDown={@continuousZoom.bind(this, .9)} onMouseUp={@toggleZoom} />
+              <button title={"zoom in"} className="zoom-in fa fa-plus" onMouseDown={@continuousZoom.bind(this, .9)} onMouseUp={@stopZoom} />
```
Comment (eatyourgreens) [inline app/components/frame-viewer.cjsx]: Ditto re. mouse event handlers.

File: app/components/frame-viewer.cjsx [hunk 3]
```diff
@@ -200,17 +197,18 @@ module.exports = React.createClass
 
   continuousZoom: (change) ->
     return if change == 0
```
Comment (eatyourgreens) [inline app/components/frame-viewer.cjsx]: Do you need to check for a timeout and clear it, before you return here?

File: app/components/frame-viewer.cjsx [hunk 4]
```diff
@@ -200,17 +197,18 @@ module.exports = React.createClass
...
-    zoomNow = =>
-      @zoom(change)
-      clearTimeout @state.zoomingTimeoutId
-      @setState zoomingTimeoutId: setTimeout(zoomNow, 200)
-    
-    zoomNow()
+    @setState zooming: true, =>
+      zoomNow = =>
+        # if !@state.zooming, we don't want to continuously call setTimeout
+        return if !@state.zooming
```
Comment (eatyourgreens) [inline app/components/frame-viewer.cjsx]: Not quite sure what this check does, since you set `state.zooming` to true before calling this function.

**Comment by simensta** [comment]
@eatyourgreens -- thanks for the feedback! Do the above changes address your concern about corresponding key event handlers?

**Comment by eatyourgreens** [comment]
Looks good to me. :+1:

**Pull request merged**